plugin/src/test/java/io/jenkins/plugins/analysis/warnings/groovy/ParserConfigurationTest.java

```diff
@@ -68,4 +68,46 @@ void shouldWarnUserIfConsoleLogScanningPermittedIsSet() {
         final FormValidation actualTrue = configuration.doCheckConsoleLogScanningPermitted(true);
         assertThat(actualTrue.kind).isEqualTo(FormValidation.Kind.WARNING);
     }
+
+    @Test
+    void shouldSaveConfigurationIfParserIsAdded() {
+        GlobalConfigurationFacade facade = mock(GlobalConfigurationFacade.class);
+        GroovyParser test_parser = new GroovyParser("1", "", "", "", "");
+
+        ParserConfiguration configuration = new ParserConfiguration(facade);
+        configuration.addParser(test_parser);
+
+        verify(facade).save();
+        assertThat(configuration.getParsers().contains(test_parser));
+    }
+
+    @Test
+    void shouldThrowIfParserExists() {
+        GlobalConfigurationFacade facade = mock(GlobalConfigurationFacade.class);
+        GroovyParser testParser = new GroovyParser("1", "", "", "", "");
+
+        ParserConfiguration configuration = new ParserConfiguration(facade);
+        configuration.addParser(testParser);
+        verify(facade).save();
+
+        assertThatIllegalArgumentException().isThrownBy(() -> configuration.addParser(testParser));
```

Comment:
When you add the ID to the message then you can add `.withMessageContaining(ID)`